airbyte-integrations/connectors/source-postgres/src/main/java/io/airbyte/integrations/source/postgres/PostgresSource.java

```diff
@@ -314,6 +314,24 @@ public List<CheckedConsumer<JdbcDatabase, Exception>> getCheckOperations(final J
       checkOperations.add(database -> {
         PostgresUtils.checkFirstRecordWaitTime(config);
       });
+
+      checkOperations.add(database -> {
+        final String userName = config.get("username").toString();
+
+        final List<JsonNode> userPrivileges = database.queryJsons(connection -> {
+          final PreparedStatement ps = connection.prepareStatement("SELECT userepl FROM pg_user WHERE usename = ?");
+          ps.setString(1, userName);
+          LOGGER.info("Verifying required privileges for user: {}", userName);
+          return ps;
+        }, sourceOperations::rowToJson);
+
+        if (!userPrivileges.get(0).get("userepl").asBoolean()) {
+          throw new RuntimeException(
```

Comment:
Can we throw a ``ConfigErrorException`` here instead? That way, even if this failure occurs during ``read()`` or ``discover()`` the user will see the same message and it will be attributed towards a configuration error

Reply:
Makes total sense! Thanks for pointing this :)

Reply:
I will also modify the other ones in `getCheckOperations`, so we make sure all the errors are marked same way.
My doubt is... is really going to to throw `ConfigErrorException` when running `read` or `discover`? AFAIK, they are different commands and they are out of the `check` function (unless `check` is called always before)

Reply:
It will protect against future changes to that codepath. 

In addition, currently for the Postgres connector ``check()`` is called from ``read()`` : https://github.com/airbytehq/airbyte/blob/master/airbyte-integrations/connectors/source-postgres/src/main/java/io/airbyte/integrations/source/postgres/PostgresSource.java#L324

Reply:
Thanks for fixing them for the other check operations!

Reply:
Awesome! Good to know, so at least we are checking in each sync if there has been any change in the user that doesn't allow to move forward with it (Thanks for looking for it ^^)

Reply:
Ideally...we should be removing this at some point, since the platform runs `check` before a sync. @subodh1810 had mention the check is required for CDC mode in the read as well (not sure exactly why) - but either way outside of the scope of this PR